ngraph/frontend/paddlepaddle/src/op/mul.cpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mul.hpp"
+#include <ngraph/opsets/opset6.hpp>
+
+namespace ngraph
+{
+    namespace frontend
+    {
+        namespace pdpd
+        {
+            namespace op
+            {
+                NamedOutputs mul(const NodeContext& node)
+                {
+                    auto x = node.get_ng_input("X");
+                    auto y = node.get_ng_input("Y");
+                    PDPD_OP_VALIDATION_CHECK(node,
+                                             x.get_partial_shape().rank().is_static(),
+                                             "matmul: X rank must be static!");
+                    int64_t x_rank = x.get_partial_shape().rank().get_length();
+                    PDPD_OP_VALIDATION_CHECK(node,
+                                             y.get_partial_shape().rank().is_static() &&
+                                                 y.get_partial_shape().rank().get_length() == 2,
+                                             "matmul: Y rank must be static, and 2!");
```

Comment:
We can get rid of static rank requirement here.  Generate code based on Rank == ShapeOf(ShapeOf). Then to take elements from the Shape -- generate the right amount of axes with Range operation. Could be taken separately -- please create ticket

Reply:
Ticket created: 61179

Reply:
@jane-intel create JIRA 6179 to refine this later.